cmd/vcluster/cmd/values/v1/base_values.go

```diff
@@ -0,0 +1,344 @@
+type RBACValues struct {
+	ClusterRole struct {
+		Create bool `json:"create,omitempty"`
+	} `json:"clusterRole,omitempty"`
+	Role struct {
+		Create               bool     `json:"create,omitempty"`
+		Extended             bool     `json:"extended,omitempty"`
+		ExcludedAPIResources []string `json:"excludedAPIResources,omitempty"`
+	} `json:"role,omitempty"`
+}
+
+type PDBValues struct {
+	Enabled bool `json:"enabled,omitempty"`
+	policyv1.PodDisruptionBudgetSpec
+}
+
+type ServiceValues struct {
+	Type                     corev1.ServiceType                  `json:"type,omitempty"`
+	ExternalIPs              []string                            `json:"externalIPs,omitempty"`
+	ExternalTrafficPolicy    corev1.ServiceExternalTrafficPolicy `json:"externalTrafficPolicy,omitempty"`
+	LoadBalancerIP           string                              `json:"loadBalancerIP,omitempty"`
+	LoadBalancerSourceRanges []string                            `json:"loadBalancerSourceRanges,omitempty"`
+	LoadBalancerClass        string                              `json:"loadBalancerClass,omitempty"`
+}
+
+type IngressValues struct {
+	Enabled          bool                      `json:"enabled,omitempty"`
+	PathType         string                    `json:"pathType,omitempty"`
+	APIVersion       string                    `json:"apiVersion,omitempty"`
+	IngressClassName string                    `json:"ingressClassName,omitempty"`
+	Host             string                    `json:"host,omitempty"`
+	Annotations      map[string]string         `json:"annotations,omitempty"`
+	TLS              []networkingv1.IngressTLS `json:"tls,omitempty"`
+}
+
+type CoreDNSValues struct {
+	Integrated     bool                        `json:"integrated,omitempty"`
+	Plugin         CoreDNSPluginValues         `json:"plugin,omitempty"`
+	Enabled        bool                        `json:"enabled,omitempty"`
+	Replicas       uint32                      `json:"replicas,omitempty"`
+	NodeSelector   corev1.NodeSelector         `json:"nodeSelector,omitempty"`
+	Image          string                      `json:"image,omitempty"`
+	Config         string                      `json:"config,omitempty"`
+	Service        CoreDNSServiceValues        `json:"service,omitempty"`
+	Resources      corev1.ResourceRequirements `json:"resources,omitempty"`
+	Manifests      string                      `json:"manifests,omitempty"`
+	PodAnnotations map[string]string           `json:"podAnnotations,omitempty"`
+	PodLabels      map[string]string           `json:"podLabels,omitempty"`
+}
+
+type CoreDNSPluginValues struct {
+	Enabled bool
+	Config  []DNSMappings
+}
+
+type DNSMappings struct {
+	Record    Record       `json:"record"`
+	Target    Target       `json:"target"`
+	AllowedOn []FilterSpec `json:"allowedOn"`
+	ExceptOn  []FilterSpec `json:"exceptOn"`
+}
+
+type Record struct {
+	RecordType RecordType
+	FQDN       *string `json:"fqdn"`
+	Service    *string `json:"service"`
```

Comment:
Are those missing the omitempty on purpose/do we want to serialize nil values explicitly?